netbox/templates/core/datasource.html

```diff
@@ -88,7 +89,18 @@ <h5 class="card-header">Backend</h5>
             {% for name, field in object.get_backend.parameters.items %}
               <tr>
                 <th scope="row">{{ field.label }}</th>
-                <td>{{ object.parameters|get_key:name|placeholder }}</td>
+                {% if field.label == "Password" and object.parameters|get_key:name %}
```

Comment:
I don't think this is going to be a viable approach. We can't rely on the field label to infer whether the parameter should be displayed. (And even if we could, hard-coding this in the template should be considered poor practice.) We should devise some approach to explicitly set some attribute on the parameter itself to control its display.